pkg/proxy/downstream.go

```diff
@@ -697,6 +709,41 @@ func (s *downStream) chooseHost(endStream bool) {
 		}
 		return
 	}
+
+	if rule := s.route.RedirectRule(); rule != nil {
```

Comment:
If both the `DirectResponseRule` and `RedirectRule` rules are configured,  it would be better that `RedirectRule` rule had a higher priority?